### ISSUE TITLE
Understanding related to implementation

### ISSUE THREAD

**Issue by ghost** [daynebatten/keras-wtte-rnn#1]
Hi, 
This post is not essentially related to issues at all. I was just thinking aloud that if I use binary variable as Response instead of single variable classification model and use sigmoid activation instead of softplus in the custom function then will the system is similar to what we called cox ph model because there also we are trying to estimate weibull parameters.

Any help will be appreciated.

**Comment by ragulpr** [comment]
In cox ph you only learn the location-parameter (alpha in the weibull) while keeping beta constant 1. Can you elaborate on the activation? Isn't ph basically exponential activation s.t the log becomes a linear combination of the input? (which in the ph layer is the input layer while in this framework is the nn layer above)

**Comment by ghost** [comment]
Yes you are absolutely right. I didn't think this way "In cox ph you only learn the location-parameter (alpha in the weibull) while keeping beta constant 1". I was actually thinking that in cox ph model our response is not of single class. I have used softplus in one of my earlier task to characterize categories which are ordinal. I could explain briefly what I had done. 
Let's say you want to predict the result of a football match. So it could be home win, draw or away win but we can not say that these are forming multinomial since these are not from two different universe like smokers and non-smokers. We will always had a single probability value, applied some threshold and obtained the outcome. Similarly here also softplus function is telling us the probability of surviving at a particular time point t. 
But if I have 0 and 1 mixed up, which is not the case for iot beacuse if engines continues to fail then only we can have lot of 0 and 1 to train with. But if I have, then do I need to tweak the activation into sigmoid or softplus will still be meaningful.

**Comment by ragulpr** [comment]
Just to try to understand what you're saying:
Football match example:

    [win, loose] x [home, away] = [win home, loose home,win away, loose away] 

Still a multinomial problem. But there is an ordinal structure like good -> bad

    [win home,win away, loose away,loose home]  

If you want to encode that it's worse to predict *win home* when it's actually *loose home* than to predict *loose away* that's a problem of missclassification cost and weighting the loss function, not about activation function. I know there's ordinal regression/glm framework but isn't that basically the thing? 

For survival problems the number of categories is typically infinite 

    [die in 1 day, die in 2 days, ....]

Hence you predict an integer or double rather than a class.

**Comment by ghost** [comment]
Ok. Now it is alright for me. Since am a newbie in survival, initially I found this one hard to map with knowledge related to CoxPh Model. 
Appreciated your response. :)
